proposals/readonly-methods.md

```diff
@@ -0,0 +1,65 @@
+# Readonly Methods
+
+* [x] Proposed
+* [ ] Prototype: Not Started
+* [ ] Implementation: Not Started
+* [ ] Specification: Not Started
+
+## Summary
+[summary]: #summary
+
+Provide a way to specify individual methods on a struct do not modify state, in the same way that `readonly struct` specifies no instance method modifies state.
+
+It is worth noting that `readonly method` != `pure method`. A `pure` method guarantees no state will be modified. A `readonly` method only guarantees that instance state will not be modified.
+
+All instance methods on a `readonly struct` could be considered implicitly `readonly methods`. Explicit `readonly methods` declared on non-readonly structs would behave in the same manner. For example, they would still create hidden copies if you called an instance method (on the current instance or on a field of the instance) which was itself not-readonly.
+
+## Motivation
+[motivation]: #motivation
+
+Today, users have the ability to create `readonly struct` types which the compiler enforces that all fields are readonly (and by extension, that no methods modify the state). However, there are some scenarios where you have an existing API that exposes public fields, or needs to modify some small amount of internal state in select scenarios. Under these circumstances, you cannot mark the type as `readonly` (it would be a breaking change).
```

Comment:
I think "expose public fields" is too limiting. Possibly use something along the lines of "exposes accessible fields and has a mix of mutating and non-mutating members".

Reply:
Fixed.